src/pages/profiles/ProfileList.tsx

```diff
@@ -1,46 +1,111 @@
-import React, { FC } from "react";
-import { MainTable, Row, Button } from "@canonical/react-components";
-import NotificationRow from "components/NotificationRow";
+import React, { FC, useEffect, useState } from "react";
+import {
+  MainTable,
+  Row,
+  Button,
+  Notification,
+  SearchBox,
+  Col,
+} from "@canonical/react-components";
 import { fetchProfiles } from "api/profiles";
-import BaseLayout from "components/BaseLayout";
+import classnames from "classnames";
 import { useQuery } from "@tanstack/react-query";
 import { queryKeys } from "util/queryKeys";
 import { useNotify } from "context/notify";
 import Loader from "components/Loader";
 import { Link, useNavigate, useParams } from "react-router-dom";
 import ItemName from "components/ItemName";
 import { getProfileInstances } from "util/usedBy";
+import usePanelParams from "util/usePanelParams";
+import { usePagination } from "util/pagination";
+import useEventListener from "@use-it/event-listener";
+import { updateTBodyHeight } from "util/updateTBodyHeight";
+import Pagination from "components/Pagination";
+import NotificationRow from "components/NotificationRow";
+import { fetchProject } from "api/projects";
 
 const ProfileList: FC = () => {
   const navigate = useNavigate();
   const notify = useNotify();
+  const panelParams = usePanelParams();
   const { project } = useParams<{ project: string }>();
+  const [query, setQuery] = useState<string>("");
 
   if (!project) {
     return <>Missing project</>;
   }
+  const isDefaultProject = project === "default";
 
   const {
     data: profiles = [],
     error,
-    isLoading,
+    isLoading: isProfilesLoading,
   } = useQuery({
     queryKey: [queryKeys.profiles, project],
     queryFn: () => fetchProfiles(project),
   });
 
+  const {
+    data: projectObj,
```

Comment:
I'm not a fan of the `*Obj` suffix. Maybe going forward, we can start aliasing the `project` from useParams to `projectName`, and have the project variable name free for the actual object.

Anyhow, not a big issue, feel free to keep it as is for now.